### ISSUE TITLE
Multiple sequences

### ISSUE THREAD

**Issue by lemuriandezapada** [fchollet/keras#69]
Hey guys,

Is this thing actually supported?:
"        Eats inputs with shape:
        (nb_samples, max_sample_length (samples shorter than this are padded with zeros at the end), input_dim)
"
The recurrent models only take (input_length, input_dim) sized inputs. Perhaps change the comments to remove this part of the description.

To deal with multiple sequences currently implies merging them into one big sequence and padding them to alignment and I see no other way around it.

**Comment by fchollet** [comment]
That is precisely how recurrent layers work, they take 3D inputs of shape (nb_samples, max_sample_length, input_dim). nb_samples is the batch size.

Can you post your model and explain what you were trying to do?

**Comment by lemuriandezapada** [comment]
Hey, so what I have is a list of strings of text of variable length. I turn them into a list of word sequences with the tokenizer. Ideally I'll turn it into a 1-of-N encoding but just for testing puposes now. I want to try to feed it one of these sequences and predict the next word. 

```python
toki.fit_on_texts(texts)
textseq = toki.texts_to_sequences(texts)
padtext = sequence.pad_sequences(textseq)
padtext = padtext.reshape(padtext.shape[0],padtext.shape[1],1) #4 1d sequences, each of length 400k
vocabsize = len(toki.word_index)

model = Sequential()
model.add(LSTM(vocabsize,128))
model.add(Dense(128, vocabsize, init='lecun_uniform'))
model.add(Activation('sigmoid'))

loss = RMSprop(lr=0.01, rho=0.9, epsilon=1e-6)
model.compile(loss='categorical_crossentropy', optimizer=loss, class_mode="categorical")
```
doing

```python
model.fit(padtext[:,:-1,:],padtext[:,:-1,:])
```

gives me the error

```
('Bad input argument to theano function with name "build/bdist.linux-x86_64/egg/keras/models.py:64"  at index 1(0-based)', 'Wrong number of dimensions: expected 2, got 3 with shape (4, 405985, 1).')
```
telling me LSTMs don't actually want 3d arrays. Ideas?

**Comment by lemuriandezapada** [comment]
It wasn't the LSTM, it was some other level in the network. Reshaping and flattening where required fixes most issues. Thanks :)
